File: ws/lomas/u3/ws24OOPConceptsZoo/OOPConceptsZoo.java/NetBeansProjects/OOPConceptsZoo/src/main/java/ec/edu/espe/oopconceptszoo/model/Pig.java
package ec.edu.espe.oopconceptszoo.model;

import ec.edu.espe.oopconceptszoo.controller.IMeatAnimal;
import java.util.ArrayList;
import java.util.Date;

public class Pig extends FarmAnimal implements IMeatAnimal {
    float idealWeight;

    public Pig(int id, String breed, Date bornOn, float weight, float idealWeight) {
        super(id, breed, bornOn, weight);
        this.idealWeight = idealWeight;
    }

    public void sendToButcher() {
        System.out.println("Pig " + id + " sent to butcher.");
    }

    @Override
    public ArrayList<Cut> cut() {
        ArrayList<Cut> cuts = new ArrayList<>();
        cuts.add(new Cut(1, "Pork Chop", "Standard Cut", 2.5f));
        cuts.add(new Cut(2, "Ribs", "Rib Extraction", 4.0f));
        return cuts;
    }

    @Override
    public void sendToSlaughterHouse(SlaughterHouse slaughterhouse) {
        System.out.println("Pig " + id + " sent to " + slaughterhouse.description);
    }

    public float getIdealWeight() { return idealWeight; }
}

File: hw/alban/u3/hw21AniamlFarm/AnimalFarm.javascript/model/Pig.js
import { FarmAnimal } from "./FarmAnimal.js";
import { Cut } from "./Cut.js";

export class Pig extends FarmAnimal {
    constructor(id, breed, bornOn, weight, idealWeight) {
        super(id, breed, bornOn, weight);
        this.idealWeight = idealWeight;
    }

    sendToButcher() {
        console.log(`Pig ${this.id} sent to butcher.`);
    }

    cut() {
        return [
            new Cut(1, "Pork Chop", "Standard Cut", 2.5),
            new Cut(2, "Ribs", "Rib Extraction", 4.0)
        ];
    }

    sendToSlaughterHouse(slaughterhouse) {
        console.log(`Pig ${this.id} sent to ${slaughterhouse.description}`);
    }
}